packages/ui/src/helpers/authenticator/utils.ts

```diff
@@ -33,7 +36,12 @@ export const censorPhoneNumber = (val: string): string => {
   return split.join('');
 };
 
-export const defaultAuthHubHandler: HubHandler = (data, service) => {
+const waitForAutoSignInState = async (service: AuthInterpreter) => {
+  // https://xstate.js.org/docs/guides/interpretation.html#waitfor
+  await waitFor(service, (state) => getActorState(state).matches('autoSignIn'));
+};
+
+export const defaultAuthHubHandler: HubHandler = async (data, service) => {
```

Comment:
Changing this handler to async, to wait for `autoSignIn` state. This is non-breaking change, because hub handler was side effects from Hub events in the first place.